internal/api/external.go

```diff
@@ -65,6 +70,32 @@ func (a *API) ExternalProviderRedirect(w http.ResponseWriter, r *http.Request) e
 	log := observability.GetLogEntry(r)
 	log.WithField("provider", providerType).Info("Redirecting to external provider")
 
+	flowStateID := ""
+	if flowType == PKCE {
+		var codeChallengeMethod models.CodeChallengeMethod
+		switch strings.ToLower(codeChallengeMethodParam) {
+		case "plain":
+			codeChallengeMethod = models.Plain
+		case "s256":
+			codeChallengeMethod = models.SHA256
+		default:
+			return badRequestError("code challenge method is unsupported")
+		}
+		codeChallenge := query.Get("code_challenge")
+		if !isValidCodeChallenge(codeChallenge) {
+			return badRequestError("invalid code challenge")
+		}
+		flowState, err := models.NewFlowState(providerType, codeChallenge, codeChallengeMethod)
+		if err != nil {
+			return err
+		}
+		if err := a.db.Create(flowState); err != nil {
+			return err
+		}
+
+		flowStateID = flowState.ID.String()
+	}
```

Comment:
What if it's not PKCE but also not empty? Can you return an error? I know I'm a bit annoying with validation, but it really does make a difference both for security and knowing why something is not working right.

Reply:
It should be a no-op but yeah agree that it'd be nice to have more validation. I'll add it in